### PR TITLE
2 rules for Azure key_vault: CR-2388 + CR-2404

### PR DESCRIPTION
**Pull request by Tomerva**

### CODE REVIEW

**Review by Viperain — APPROVED** [review]

**Review by yi2020 — APPROVED** [review]

File: cloudrail/knowledge/rules/azure/context_aware/key_vault_diagnostic_logs_enabled_rule.py [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+        for key_vault in env_context.key_vaults:
+            if \
+                    not key_vault.monitor_diagnostic_settings \
+                    or not key_vault.monitor_diagnostic_settings.logs_settings \
+                    or not key_vault.monitor_diagnostic_settings.logs_settings.enabled \
+                    or not key_vault.monitor_diagnostic_settings.logs_settings.retention_policy \
+                    or not key_vault.monitor_diagnostic_settings.logs_settings.retention_policy.enabled:
+                issues.append(
+                    Issue(
+                        f'The Key Vault `{key_vault.name}` does not have diagnostic logs enabled.',
```
Comment (yi2020) [inline cloudrail/knowledge/rules/azure/context_aware/key_vault_diagnostic_logs_enabled_rule.py:24]: Looking at Oleg's comments on CR-2388, we need to be clear here what's not enabled. For example, user may have enabled logs setting but not retention policy.

**Review by yi2020 — CHANGES_REQUESTED** [review]

File: cloudrail/knowledge/context/azure/keyvault/azure_monitor_diagnostic_setting.py [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+    enabled: bool
+
+
+@dataclass
+class AzureMonitorDiagnosticLogsSettings:
+    enabled: bool
+    retention_policy: AzureMonitorDiagnosticLogsRetentionPolicySettings
+
+
+class AzureMonitorDiagnosticSetting(AzureResource):
```
Comment (Tomerva) [inline cloudrail/knowledge/context/azure/keyvault/azure_monitor_diagnostic_setting.py:19]: add is_standalone implementation

File: tests/knowledge/rules/azure/context_aware/test_key_vault_diagnostic_logs_enabled.py [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+
+
+class TestKeyVaultDiagnosticLogsEnabled(unittest.TestCase):
+    def setUp(self):
+        self.rule = KeyVaultDiagnosticLogsEnabledRule()
+
+    @parameterized.expand(
+        [
+            [None, True],
+            [AzureMonitorDiagnosticSetting('settings_name', 'keyvault_id', None), True],
```
Comment (Tomerva) [inline tests/knowledge/rules/azure/context_aware/test_key_vault_diagnostic_logs_enabled.py:22]: add name

File: cloudrail/knowledge/context/azure/keyvault/azure_monitor_diagnostic_setting.py [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+        Attributes:
+            name: The monitor diagnostic setting's name
+            monitoring: The ID of the resource that is monitored
+            logs_settings: The logs settings
+    """
+
+    def __init__(self, name: str, monitoring: str, logs_settings: Optional[AzureMonitorDiagnosticLogsSettings]):
+        super().__init__(AzureResourceType.AZURERM_MONITOR_DIAGNOSTIC_SETTING)
+        self.name: str = name
+        self.monitoring: str = monitoring
```
Comment (Tomerva) [inline cloudrail/knowledge/context/azure/keyvault/azure_monitor_diagnostic_setting.py:30]: rename to target_resource_id

**Review by alonashkenazi — APPROVED** [review]

File: cloudrail/knowledge/context/azure/azure_resource.py [hunk 5]
```diff
@@ -38,4 +38,6 @@ def get_id(self) -> str:
         return self._id
 
     def set_id(self, _id: str):
-        self._id = _id
+        if _id:
+            self._id = _id
+            self.with_aliases(_id)
```
Comment (yi2020) [inline cloudrail/knowledge/context/azure/azure_resource.py:43]: This is something @Tomerva should look at

Comment (Tomerva) [reply cloudrail/knowledge/context/azure/azure_resource.py:43]: this is my change

**Review by yi2020 — APPROVED** [review]

**Pull request merged**